File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.java
import java.util.Arrays;

class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        Arrays.sort(happiness);

        int i = 0;
        long res = 0;
        int n = happiness.length;

        while (k > 0 && i < n) {
            int index = n - 1 - i;
            int adjustedHappiness = Math.max(happiness[index] - i, 0);
            res += adjustedHappiness;
            i++;
            k--;
        }
        return res;
    }
}

File: 3075-maximize-happiness-of-selected-children/3075-maximize-happiness-of-selected-children.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort()
        res = 0
        count = 0
        for i in range(len(happiness) - 1, len(happiness) - k - 1, -1):
            if happiness[i] - count > 0:
                res += (happiness[i] - count)
            count += 1
        return res